### PR TITLE
[Presentation] Automate contract binding build

### PR DESCRIPTION
**Pull request by ChihChengLiang**
### What's wrong

- We manually copy paste abi from the hubble-contracts.
- We don't know what version of ABI that the contract binding is using.
- We need to pave the way for the coordinator client testing

### How are we fixing it?

- Create a submodule of hubble-contract. So we can know hubble-commander is compatible with the what version of  hubble-contract.
- Automate the contract binding generation from the hubble-contract submodule. Manual copy-pasting is no longer required.
- Add a check in CI so that we don't forget to update the ABI binding when we upgrade the submodule.

### CODE REVIEW

**Comment by ChihChengLiang** [comment]
@vaibhavchellani Sorry for the huge diff. `add generate script and update Makefile ci` is the real commit to review

File: Makefile [hunk 1]
```diff
@@ -1,14 +1,9 @@
...
-	abigen --abi=contracts/registry/registry.abi --pkg=registry --out=contracts/registry/registry.go
-	abigen --abi=contracts/state/state.abi --pkg=state --out=contracts/state/state.go
-	abigen --abi=contracts/depositmanager/depositmanager.abi --pkg=depositmanager --out=contracts/depositmanager/depositmanager.go
-	abigen --abi=contracts/tokenregistry/tokenregistry.abi --pkg=tokenregistry --out=contracts/tokenregistry/tokenregistry.go
-	abigen --abi=contracts/burnauction/burnauction.abi --pkg=burnauction --out=contracts/burnauction/burnauction.go
-	abigen --abi=contracts/transfer/transfer.abi --pkg=transfer --out=contracts/transfer/transfer.go
-	abigen --abi=contracts/massmigration/massmigration.abi --pkg=massmigration --out=contracts/massmigration/massmigration.go
-	abigen --abi=contracts/create2transfer/create2transfer.abi --pkg=create2transfer --out=contracts/create2transfer/create2transfer.go
-	abigen --abi=contracts/accountregistry/accountregistry.abi --pkg=accountregistry --out=contracts/accountregistry/accountregistry.go
+build-hubble-contracts:
```
Comment (vaibhavchellani) [inline Makefile:1]: This introduces a `nodejs` requirement for `commander` , right? Is there any way to avoid it?

Comment (ChihChengLiang) [reply Makefile:1]: only when we need to generate the abi binding, which happens when we bump contract submodule version.
Building commander doesn't require nodejs.

Comment (vaibhavchellani) [reply Makefile:1]: Nice!

**Review by vaibhavchellani — APPROVED** [review]
LGTM!

**Pull request merged**